fiftyone/core/dataset.py

```diff
@@ -2569,6 +2569,7 @@ def _add_samples_batch(
                 sample.frames.save()
 
         if batcher is not None and batcher.manual_backpressure:
+            # @todo can we infer content size from insert_many() above?
```

Comment:
What do you mean, we can't do much better than getting the sum of BSON representations here.

Reply:
Just hoping/wondering if somehow the request size could be *inferred* from the `insert_many()` call above rather than manually recomputing it here 🤷 

Serializing large docs can be a significant cost

Reply:
Ah yes unfortunately not sure that we can, the return result doesn't give much information.
We could sample the docs instead of summing all of them.

But the thinking was that if you need to use the content size batcher it's because you have a higher overhead on each call (Teams use case), and therefore the cost of serializing the docs twice instead of once is dwarfed by the overhead of the call itself so it's sort of negligible.

Reply:
> But the thinking was that if you need to use the content size batcher it's because you have a higher overhead on each call (Teams use case), and therefore the cost of serializing the docs twice instead of once is dwarfed by the 
overhead of the call itself so it's sort of negligible

Yep agree with this 100%, which is why I left the @todo rather than actually trying to address them now. But if there's minimal hope of achieving the @todo dreams we can just delete.

Adding optional support for random sampling to `ContentSizeDynamicBatcher` would be a nice addition 👍

Reply:
I'm thinking there's minimal hope but sure we can leave it in, in case someone has an epiphany later (or pymongo adds info to return types)